subsys/net/lib/lwm2m/lwm2m_rw_oma_tlv.c

```diff
@@ -903,6 +903,51 @@ static int write_tlv_resource(struct lwm2m_message *msg, struct oma_tlv *tlv)
 	return 0;
 }
 
+static int lwm2m_multi_resource_tlv_parse(struct lwm2m_message *msg,
+					  struct oma_tlv *multi_resource_tlv)
+{
+	struct oma_tlv tlv_resource_instance;
+	int len2;
+	int pos = 0;
+	int ret;
+
+	if (msg->in.block_ctx) {
+		msg->in.block_ctx->res_id = multi_resource_tlv->id;
+	}
+
+	if (multi_resource_tlv->length == 0U) {
+		/* No data for resource instances, so create only a resource */
+		return write_tlv_resource(msg, multi_resource_tlv);
+	}
+
+	while (pos < multi_resource_tlv->length &&
+	       (len2 = oma_tlv_get(&tlv_resource_instance, &msg->in, true))) {
+		if (tlv_resource_instance.type != OMA_TLV_TYPE_RESOURCE_INSTANCE) {
+			LOG_ERR("Multi resource id not supported %u %d", tlv_resource_instance.id,
+				tlv_resource_instance.length);
+			return -ENOTSUP;
+		}
+
+		msg->path.res_id = multi_resource_tlv->id;
+		msg->path.res_inst_id = tlv_resource_instance.id;
+		msg->path.level = 4U;
```

Comment:
nit: Would be nice to use symbols instead of magic numbers (`LWM2M_PATH_LEVEL_RESOURCE_INST`).